cal/cal/static/js/app.js

```diff
@@ -102,33 +102,125 @@ analyticsApp.controller('TagsCtrl', function($scope, $http){
+  $scope.categories = [];
+  $scope.categoriesForChart = [];
 
-  // Generate graph data
-  $http({ method: 'GET', url: url }).
-    success(function (data) {
-      // set the data
-      $scope.categories = [];
+  // populate the categories pie chart
+  $http({ method: 'GET', url: categoryUrl + '.json' }).
+    success(function successCallback(data) {
       for (var i = 0; i < data.results.length; i++) {
         var category = data.results[i];
         $scope.categories.push({
+          id: category.id,
+          label: category.label,
+          hours: category.hours,
+          include: true
+        });
+        $scope.categoriesForChart.push({
+          id: category.id,
           label: category.label,
-          hours: category.hours
+          hours: category.hours,
+          include: true
         });
       }
     });
 
+  this.startEdit = function(categoryId) {
+    var category = $scope.categories.find(function(category, index, array) {
+      return category.id == categoryId;
+    });
+    category.newLabel = category.label;
+    category.editing = true;
+  };
+
+  this.submit = function(categoryId) {
+    var category = $scope.categories.find(function(category, index, array) {
+      return category.id == categoryId;
+    });
+    category.editing = false;
+
+    $http({
+      method: 'POST',
+      url: categoryUrl + '/' + categoryId,
+      data: $.param({
+        label: category.newLabel,
+        csrfmiddlewaretoken: getCookie('csrftoken'),
+        _method: 'PATCH'
+      }),
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded'
+      }
+    }).
+      success(function addToList(data) {
+        category.label = data.label;
+        category.hours = data.hours;
+      });
+  };
+
+
+  this.cancelEdit = function(categoryId) {
+    var category = $scope.categories.find(function(category, index, array) {
+      return category.id == categoryId;
+    });
+    category.editing = false;
+  };
+
+  this.remove = function(categoryId) {
+    $http({
+      method: 'POST',
+      url: categoryUrl + '/' + categoryId,
+      data: $.param({
+        csrfmiddlewaretoken: getCookie('csrftoken'),
+        _method: 'DELETE'
+      }),
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded'
+      }
+    }).
+      success(function removeFromList(data) {
+        $scope.categories = $scope.categories.filter(function(category) {
+          return category.id !== categoryId;
+        });
+      });
+  };
+
+  // when category is checked, include in pie chart, if not don't include
+  this.chartToggle = function(categoryId) {
+    var category = $scope.categories.find(function(category, index, array) {
+      return category.id == categoryId;
+    });
+
+    // when user checks a category
+    if (category.include) {
+      $scope.categoriesForChart.push({
+        id: category.id,
+        label: category.label,
+        hours: category.hours,
+      });
+      category.include = true;
```

Comment:
I don't think this line of code is working the way you expect it to. In line 198 we have an if statement asking if `category.include` is `true`, and since line 204 is in this if block set it to true, it's redundant because `category.include` should already be set to true.

You can delete this whole if-else block though, since we don't need the `categoriesForChart` variable.